pkg/autodiscovery/common/utils/kubelet.go

```diff
@@ -16,3 +20,35 @@ func GetCustomCheckID(annotations map[string]string, containerName string) (stri
 	id, found := annotations[fmt.Sprintf(checkIDAnnotationFormat, containerName)]
 	return id, found
 }
+
+// ValidateAnnotationsMatching detects if AD annotations don't match a valid container identifier
+func ValidateAnnotationsMatching(annotations map[string]string, containerIdentifiers map[string]bool, containerNames map[string]bool, adPrefix string) []error {
+	var errors []error
+	adAnnotation := fmt.Sprintf(`%s.+\..+`, adPrefix)
+	checkIDAnnotation := fmt.Sprintf(checkIDAnnotationFormat, ".+\\")
+
+	for annotation := range annotations {
+		if matched, _ := regexp.MatchString(checkIDAnnotation, annotation); matched {
+			// validate check.id annotation
+			err := validateIdentifier(annotation, containerNames, adPrefix)
+			if err != nil {
+				errors = append(errors, err)
+			}
+		} else if matched, _ := regexp.MatchString(adAnnotation, annotation); matched {
```

Comment:
What do you think about using `strings.HasPrefix` here instead of a regular expression?

Reply:
I'm worried that `strings.HasPrefix` would also match an annotation like `ad.datadoghq.com/tags`, which should be ignored.

Reply:
You are right, the second part of the annotation would need to be checked separately. But note that in the latest revision you already calling `LastIndex`, it could be also used to check if the part after the prefix contains the dot.